app/jobs/start_sync_job.rb

```diff
@@ -22,8 +22,20 @@ def perform(sync_id, force: false)
 
   private
 
-  def imports_to_sync?
-    Import.where(state: ImportMicroMachine::FINISHED).any?
+  def should_sync?
+    finished_imports = Import.where(state: ImportMicroMachine::FINISHED)
+
+    return false unless finished_imports
+
+    imports_to_sync = finished_imports.count
+
+    finished_imports.each do |import|
+      errors_count = import.raw_inputs.where('exception is not null').count
+      next unless import.raw_inputs.count == errors_count
+      imports_to_sync -= 1
```

Comment:
Cool, I like this! My only thought is to move the knowledge around a little. What if we added a method to Import called `every_raw_inputs_failed`:

```ruby
class Import
  def every_raw_inputs_failed?
    raw_inputs.where.not(exception: nil).count == raw_inputs.count
  end
end
```

Then, the worker could just do something like this:

```ruby
def should_sync?
  finished_imports = Import.where(state: ImportMicroMachine::FINISHED)
  finished_imports.all?(&:every_raw_inputs_failed?)
end
```

I realize `every_raw_inputs_failed?` is pretty long, def open to ideas there.

I think a refactor like this is nice so that we can keep the worker ignorant about the details of what makes an import something that doesn't need to be synced.